.ci/Jenkinsfile

```diff
@@ -337,25 +339,42 @@ def runBenchmark(version){
 }
 
 def runJob(String rubyVersion, int sleep = 1){
+  def buildFailed = false
   def buildObject
+  def buildNumber
   def branch = env.CHANGE_ID?.trim() ? env.CHANGE_TARGET : env.BRANCH_NAME
+  def jobName = "apm-agent-ruby/apm-agent-ruby-downstream/${env.BRANCH_NAME}"
   try {
     def quiet = (sleep * randomNumber(min: 2, max: 6)) + 5
-    buildObject = build(job: "apm-agent-ruby/apm-agent-ruby-downstream/${env.BRANCH_NAME}",
+    buildObject = build(job: jobName,
                         parameters: [
                           string(name: 'RUBY_VERSION', value: "${rubyVersion}"),
                           string(name: 'BRANCH_SPECIFIER', value: "${env.GIT_BASE_COMMIT}"),
                           string(name: 'MERGE_TARGET', value: "${branch}")
                         ],
                         quietPeriod: quiet)
+    buildNumber = buildObject.number.toString()
   } catch(e) {
     rubyTasksFailed = true
+    buildFailed = true
+    buildNumber = buildObject.number.toString()
+    log(level: 'INFO', text: "Job failed '${buildObject?.name}' - ${buildNumber}")
     buildObject = e
-    warnError('Downstream Failed') {
-      error("Downstream job for '${rubyVersion}' failed")
-    }
   } finally {
+    dir(rubyVersion) {
+      copyArtifacts(projectName: jobName, filter: '**/ruby-agent-junit.xml', selector: specific(buildNumber: buildNumber))
+    }
```

Comment:
Using the rubyVersion to help with the categorisation of the tests, so the same stage will run for different downstreams, so this will enhance the test reporting:

![image](https://user-images.githubusercontent.com/2871786/81085480-2bb79180-8eef-11ea-8548-321a2b17b62b.png)